File: DemoObjC/DemoObjC/View Controllers/Adapter - AppLovin MAX/MAXBannerAdVC.m
//
//  MAXBannerAdVC.m
//  DemoObjC
//
// © 2026 Limelight Inc. All Rights Reserved.
//

#import "MAXBannerAdVC.h"
#import <AppLovinSDK/AppLovinSDK.h>
#import <LimelightSDK/LimelightSDK.h>
#import <LimelightSDKAppLovinAdapter/LimelightSDKAppLovinAdapter.h>

// MARK: - Private Interface

@interface MAXBannerAdVC ()

// MARK: Storyboard UI

@property (nonatomic, weak) IBOutlet UIView *adContainerView;
// Ad Info
@property (nonatomic, weak) IBOutlet UILabel *adUnitIdLabel;
@property (nonatomic, weak) IBOutlet UILabel *adSizeLabel;
// MAAdViewAdDelegate
@property (nonatomic, weak) IBOutlet UILabel *didLoadLabel;
@property (nonatomic, weak) IBOutlet UILabel *didFailToLoadAdLabel;
@property (nonatomic, weak) IBOutlet UILabel *didDisplayLabel;
@property (nonatomic, weak) IBOutlet UILabel *didHideLabel;
@property (nonatomic, weak) IBOutlet UILabel *didClickLabel;
@property (nonatomic, weak) IBOutlet UILabel *didFailToDisplayLabel;
@property (nonatomic, weak) IBOutlet UILabel *didExpandLabel;
@property (nonatomic, weak) IBOutlet UILabel *didCollapseLabel;
// Controls
@property (nonatomic, weak) IBOutlet UIButton *refreshButton;
@property (nonatomic, weak) IBOutlet UIButton *stopAutoRefreshButton;

- (IBAction)didClickRefreshButton:(id)sender;
- (IBAction)didClickStopAutoRefreshButton:(id)sender;

// MARK: Ad View

@property (nonatomic, strong, nullable) MAAdView *maAdView;

@end

// MARK: - Implementation

@implementation MAXBannerAdVC

// MARK: UIViewController Lifecycle

- (void)viewDidLoad {
    [super viewDidLoad];
    [self setupAndLoadAd];
}

// MARK: Actions

- (IBAction)didClickRefreshButton:(id)sender {
    [self refreshAd];
}

- (IBAction)didClickStopAutoRefreshButton:(id)sender {
    [self stopAutoRefresh];
}

- (void)setupAndLoadAd {
    self.adUnitIdLabel.text = self.adConfigId;
    self.adSizeLabel.text = [NSString stringWithFormat:@"%dx%d", (int)self.adSize.width, (int)self.adSize.height];
    
    MAAdView *maAdView = [[MAAdView alloc] initWithAdUnitIdentifier:self.adConfigId
                                                           adFormat:self.adFormat
                                                      configuration:nil];
    maAdView.delegate = self;
    maAdView.frame = CGRectMake(0, 0, self.adSize.width, self.adSize.height);
    
    if (self.isAutoRefreshEnabled) {
        [maAdView startAutoRefresh];
    } else {
        // Set this extra parameter to work around SDK bug that ignores calls to stopAutoRefresh()
        [maAdView setExtraParameterForKey:@"allow_pause_auto_refresh_immediately" value:@"true"];
        [maAdView stopAutoRefresh];
    }
    
    if (self.isAdaptive) {
        [maAdView setExtraParameterForKey:@"adaptive_banner" value:@"true"];
    }
    
    // Add as a subview
    maAdView.translatesAutoresizingMaskIntoConstraints = NO;
    [self.adContainerView addSubview:maAdView];
    
    NSLayoutConstraint *widthConstraint = [self.adContainerView.widthAnchor constraintEqualToAnchor:maAdView.widthAnchor multiplier:1];
    NSLayoutConstraint *heightConstraint = [self.adContainerView.heightAnchor constraintEqualToAnchor:maAdView.heightAnchor multiplier:1];
    [self.adContainerView addConstraints:@[widthConstraint, heightConstraint]];
    
    self.maAdView = maAdView;
    [maAdView loadAd];
}

- (void)resetEvents {
    self.didLoadLabel.enabled = NO;
    self.didFailToLoadAdLabel.enabled = NO;
    self.didDisplayLabel.enabled = NO;
    self.didHideLabel.enabled = NO;
    self.didClickLabel.enabled = NO;
    self.didFailToDisplayLabel.enabled = NO;
    self.didExpandLabel.enabled = NO;
    self.didCollapseLabel.enabled = NO;
}

- (void)refreshAd {
    [self resetEvents];
    self.refreshButton.enabled = NO;
    [self.maAdView loadAd];
}

- (void)stopAutoRefresh {
    self.stopAutoRefreshButton.enabled = NO;
    [self.maAdView stopAutoRefresh];
}

// MARK: MAAdViewAdDelegate

- (void)didLoadAd:(MAAd *)ad {
    self.adSizeLabel.text = [NSString stringWithFormat:@"%dx%d", (int)ad.size.width, (int)ad.size.height];
    self.didLoadLabel.enabled = YES;
}

- (void)didFailToLoadAdForAdUnitIdentifier:(NSString *)adUnitIdentifier withError:(MAError *)error {
    self.didFailToLoadAdLabel.enabled = YES;
}

- (void)didDisplayAd:(MAAd *)ad {
    self.didDisplayLabel.enabled = YES;
}

- (void)didHideAd:(MAAd *)ad {
    self.didHideLabel.enabled = YES;
}

- (void)didClickAd:(MAAd *)ad {
    self.didClickLabel.enabled = YES;
}

- (void)didFailToDisplayAd:(MAAd *)ad withError:(MAError *)error {
    self.didFailToDisplayLabel.enabled = YES;
}

- (void)didExpandAd:(MAAd *)ad {
    self.didExpandLabel.enabled = YES;
}

- (void)didCollapseAd:(MAAd *)ad {
    self.didCollapseLabel.enabled = YES;
}

@end

File: DemoSwift/DemoSwift/View Controllers/Adapter - AppLovin MAX/MAXBannerAdVC.swift
//
//  MAXBannerAdVC.swift
//  DemoSwift
//
// © 2026 Limelight Inc. All Rights Reserved.
//

import AppLovinSDK
import LimelightSDK
import LimelightSDKAppLovinAdapter
import UIKit

// MARK: - MAXBannerAdVC
final class MAXBannerAdVC: UIViewController {
    
    // MARK: Storyboard UI

    @IBOutlet var adContainerView: UIView!
    // Ad Info
    @IBOutlet var adUnitIdLabel: UILabel!
    @IBOutlet var adSizeLabel: UILabel!
    // MAAdViewAdDelegate
    @IBOutlet var didLoadLabel: UILabel!
    @IBOutlet var didFailToLoadAdLabel: UILabel!
    @IBOutlet var didDisplayLabel: UILabel!
    @IBOutlet var didHideLabel: UILabel!
    @IBOutlet var didClickLabel: UILabel!
    @IBOutlet var didFailToDisplayLabel: UILabel!
    @IBOutlet var didExpandLabel: UILabel!
    @IBOutlet var didCollapseLabel: UILabel!
    // Controls
    @IBOutlet var refreshButton: UIButton!
    @IBOutlet var stopAutoRefreshButton: UIButton!
    @IBAction func didClickRefreshButton(_ sender: Any) { refreshAd() }

    @IBAction func didClickStopAutoRefreshButton(_ sender: Any) { stopAutoRefresh() }

    // MARK: Ad View

    var maAdView: MAAdView?

    // Ad View Customizations
    var adConfigId = "1234567890"
    var adSize: CGSize = .zero
    var adFormat: MAAdFormat = .banner
    var isAutoRefreshEnabled: Bool = true
    var isAdaptive: Bool = false

    // MARK: UIViewController Lifecycle

    override func viewDidLoad() {
        super.viewDidLoad()
        setupAndLoadAd()
    }
    
    // MARK: Actions
    
    func setupAndLoadAd() {
        adUnitIdLabel.text = adConfigId
        adSizeLabel.text = Int(adSize.width).description + "x" + Int(adSize.height).description

        let maAdView = MAAdView(adUnitIdentifier: adConfigId, adFormat: adFormat, configuration: nil)
        maAdView.delegate = self
        maAdView.frame = CGRect(origin: .zero, size: adSize)

        if isAutoRefreshEnabled {
            maAdView.startAutoRefresh()
        } else {
            // Set this extra parameter to work around SDK bug that ignores calls to stopAutoRefresh()
            maAdView.setExtraParameterForKey("allow_pause_auto_refresh_immediately", value: "true")
            maAdView.stopAutoRefresh()
        }

        if isAdaptive {
            maAdView.setExtraParameterForKey("adaptive_banner", value: "true")
        }

        // Add as a subview
        maAdView.translatesAutoresizingMaskIntoConstraints = false
        adContainerView.addSubview(maAdView)
        let widthConstraint = adContainerView.widthAnchor.constraint(equalTo: maAdView.widthAnchor, multiplier: 1)
        let heightConstraint = adContainerView.heightAnchor.constraint(equalTo: maAdView.heightAnchor, multiplier: 1)
        adContainerView.addConstraints([widthConstraint, heightConstraint])

        self.maAdView = maAdView

        maAdView.loadAd()
    }
    
    func resetEvents() {
        didLoadLabel.isEnabled = false
        didFailToLoadAdLabel.isEnabled = false
        didDisplayLabel.isEnabled = false
        didHideLabel.isEnabled = false
        didClickLabel.isEnabled = false
        didFailToDisplayLabel.isEnabled = false
        didExpandLabel.isEnabled = false
        didCollapseLabel.isEnabled = false
    }
    
    func refreshAd() {
        resetEvents()
        refreshButton.isEnabled = false
        maAdView?.loadAd()
    }
    
    func stopAutoRefresh() {
        stopAutoRefreshButton.isEnabled = false
        maAdView?.stopAutoRefresh()
    }
}

// MARK: - MAAdViewAdDelegate
extension MAXBannerAdVC: MAAdViewAdDelegate {
    func didLoad(_ ad: MAAd) {
        adSizeLabel.text = Int(ad.size.width).description + "x" + Int(ad.size.height).description
        didLoadLabel.isEnabled = true
    }

    func didFailToLoadAd(forAdUnitIdentifier adUnitIdentifier: String, withError error: MAError) {
        didFailToLoadAdLabel.isEnabled = true
    }

    func didDisplay(_ ad: MAAd) {
        didDisplayLabel.isEnabled = true
    }

    func didHide(_ ad: MAAd) {
        didHideLabel.isEnabled = true
    }

    func didClick(_ ad: MAAd) {
        didClickLabel.isEnabled = true
    }

    func didFail(toDisplay ad: MAAd, withError error: MAError) {
        didFailToDisplayLabel.isEnabled = true
    }

    func didExpand(_ ad: MAAd) {
        didExpandLabel.isEnabled = true
    }

    func didCollapse(_ ad: MAAd) {
        didCollapseLabel.isEnabled = true
    }
}
